codepropertygraph/src/main/scala/io/shiftleft/utils/ExecutionContextSummoning.scala

```diff
@@ -0,0 +1,135 @@
+package io.shiftleft.utils
+
+import org.slf4j.{LoggerFactory, MDC}
+
+import java.util.concurrent.{ForkJoinPool, ForkJoinWorkerThread, Semaphore, ThreadFactory}
+import scala.concurrent.{BlockContext, CanAwait, ExecutionContext, ExecutionContextExecutorService}
+
+object ExecutionContextSummoning {
+  def summonExecutionContext: ExecutionContext = {
+    Thread.currentThread() match {
+      case fjt: ForkJoinWorkerThread =>
+        fjt.getPool match {
+          case ec: ExecutionContext => ec
+          case _                    => ExecutionContext.global
+        }
+      case _ => ExecutionContext.global
+    }
+  }
+
+  val logger = LoggerFactory.getLogger(getClass)
+
+  //adapted from  scala/concurrent/impl/ExecutionContextImpl.scala
```

Comment:
How about moving this comment right above `MDCThreadFactory`? For me it wasn't obvious that the comment referred to `MDCThreadFactory`. I would also add github link to the code, I believe it is [MDCThreadFactory](https://github.com/scala/scala/blob/2.13.x/src/library/scala/concurrent/impl/ExecutionContextImpl.scala#L27) ?